modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/SidebarPanel.js

```diff
@@ -34,38 +35,49 @@ const SidebarPanel = React.forwardRef(
 		const [isOpen, setIsOpen] = useState(true);
 		const [resourceData, setResourceData] = useState();
 
-		const getData = useCallback((fetchURL) => {
-			setIsLoading(true);
-			setResourceData(null);
+		const delay = useTimeout();
 
-			fetch(fetchURL, {
-				method: 'GET',
-			})
-				.then((response) => response.json())
-				.then((data) => {
-					setData(data, data?.error);
+		const getData = useCallback(
+			(fetchURL) => {
+				setIsLoading(true);
+				setResourceData(null);
+
+				fetch(fetchURL, {
+					method: 'GET',
 				})
-				.catch(() => {
-					setData(
-						null,
-						Liferay.Language.get('an-unexpected-error-occurred')
-					);
-				});
-		}, []);
+					.then((response) =>
+						response.headers.get('content-type').includes('json')
+							? response
+									.json()
+									.then((data) => setData(data, data?.error))
+							: response.text().then((html) => setData({html}))
+					)
+					.catch(() => {
+						setData(
+							null,
+							Liferay.Language.get('an-unexpected-error-occurred')
+						);
+					});
+			},
+			[setData]
+		);
 
 		const onCloseHandle = () => (onClose ? onClose() : setIsOpen(false));
 
-		const setData = (data, error) => {
+		const setData = useCallback(
```

Comment:
(Tangential rambling.)

I know the linter told you to put this here, but you don't really need this `useCallback`, because [the `delay()` function returned by `useTimeout()`](https://github.com/liferay/liferay-portal/blob/9a9b3d1bc8199ef517d1ea94ee4158ba3ae4aeda/modules/apps/frontend-js/frontend-js-react-web/src/main/resources/META-INF/resources/js/hooks/useTimeout.es.js) already uses `useCallback` internally.

Which is to say, the identity of the `setData` function is going to be stable. Ditto for `getData()` (which I also know the linter told you to wrap in `useCallback`); it's always going to be stable.

So, you *could* remove the `useCallback` wrappers and change:

```js
useEffect(() => {
    getData(fetchURL);
}, [fetchURL, getData]);
```

to:

```js
useEffect(() => {
    getData(fetchURL);
    // eslint-disable-next-line react-hooks/exhaustive-deps
}, [fetchURL]); 
```

🤷 it's either the overhead and clutter of two useless wrappers, or the ugliness of a "trust me, I know what I'm doing" lint suppression. Probably doesn't make a big difference in the end anyway.

Reply:
Done!